.github/workflows/publish.yml

```diff
@@ -25,10 +28,18 @@ jobs:
           npm_token: ${{ secrets.NPM_TOKEN }}
           skip_build: 'true'
       - name: Build packages
-        run: | 
+        run: |
           . venv/bin/activate
-          pip install build
+          pip install build hatch
           pnpm --filter @gradio/client --filter @gradio/lite build
+      - name: Package Lite NPM package
```

Comment:
It might be better if the NPM package created in this step will be published in the following step as it's guaranteed that the NPM-published file is exactly same as what we can check from the uploaded artifacts?

Reply:
Aren't we already building part of this in the `install-deps` workflow above? 

We should keep the builds together where possible, I think moving this tot he install deps action would make the most sense. Then we can have access to these assets in a variety of workflows if needed.

Reply:
Oh the `install-frontend-deps` action has this part building Lite 👀  
You mean we should move `pnpm pack` to this?
https://github.com/gradio-app/gradio/blob/024b44c553e73f1d6c00d4c8cf12ae38181a454d/.github/actions/install-frontend-deps/action.yml#L54-L60

(Is it ok that the "install-*-deps" action is also responsible for building things which sounds not intuitive?)